.github/workflows/snap.yaml

```diff
@@ -0,0 +1,120 @@
+name: Build OpenSearch from source and run tests
+
+env:
+  VERSION: 2.2.1
+  RELEASE: edge
+
+on:
+  push:
+    branches:
+      - edge
+
+jobs:
+  build:
+    name: Build Snap
+    runs-on: ubuntu-latest
+    outputs:
+      snap-file: ${{ steps.build-snap.outputs.snap }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: edge
+
+      - id: build-snap
+        name: Build snap
+        uses: snapcore/action-build@v1
+        with:
+          snapcraft-channel: 7.x/candidate
+
+      - name: Upload built snap job artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: opensearch_snap_amd64
+          path: opensearch_${{env.VERSION}}_amd64.snap
+
+  test:
+    name: Test Snap
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    steps:
+      - name: Download snap file
+        uses: actions/download-artifact@v3
+        with:
+          name: opensearch_snap_amd64
+          path: .
+
+      - name: Install snap file
+        run: |
+          sudo snap install opensearch_${{env.VERSION}}_amd64.snap --dangerous
+
+      - name: Setup and Start OpenSearch
+        run: |
+          # connect interfaces
+          sudo snap connect opensearch:log-observe
```

Comment:
have you requested auto connection for these?

Reply:
Yes, (can be found [here](https://forum.snapcraft.io/t/auto-connect-request-for-opensearch-snap/31746)) waiting for the approval